src/powershell-native/nativemsh/pwrshcommon/ConfigFileReader.cpp

```diff
@@ -44,30 +44,42 @@ namespace NativeMsh
                 }
                 else if (*iter == L'p' || *iter == L'P')
                 {
-                    this->pathToPowerShellAssemblies = this->getValueFromLine(line, psHomeDirTag);
-                    if (this->pathToPowerShellAssemblies.size() > 0) // Found a match
+                    std::wstring psHomeDir = this->getValueFromLine(line, psHomeDirTag);
+                    HANDLE dirHandle = CreateFileW(psHomeDir.c_str(), GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
+                    if (INVALID_HANDLE_VALUE != dirHandle)
                     {
-                        std::wstring::const_iterator slashIter = this->pathToPowerShellAssemblies.end();
-                        slashIter--;
-                        if (*slashIter != L'\\')
+                        CloseHandle(dirHandle);
+                        this->pathToPowerShellAssemblies = psHomeDir;
+                        if (this->pathToPowerShellAssemblies.size() > 0) // Found a match
```

Comment:
It seems the `size() > 0` check should be moved to before the `CreateFileW` call.
Or maybe the `size() > 0` is not needed, because if `getValueFromLine` returns an empty string, I thinkg `CreateFileW` will fail.

Reply:
Wrote a small program to test this and an empty string does return an invalid handle.  I can remove the size check.